player/tests.py

```diff
@@ -97,3 +97,30 @@ def test_create_player_with_null_favorite_ball_game(self):
                 birth_date='1990-01-01',
                 favorite_ball_game=None,
             )
+
+    def test_validate_and_save_valid_data(self, player):
+        birth_date = '1999-01-01'
+        favorite_ball_game = BallGame.Soccer
+
+        player.validate_and_save(birth_date, favorite_ball_game)
+
+        assert player.birth_date.strftime('%Y-%m-%d') == birth_date
+        assert player.favorite_ball_game == favorite_ball_game
+
+    def test_validate_and_save_invalid_birth_date_format(self, player):
+        birth_date = '01-01-1999'
+        favorite_ball_game = BallGame.Basketball
+
+        with pytest.raises(ValidationError) as e:
+            player.validate_and_save(birth_date, favorite_ball_game)
+
+        assert "Invalid birth date format. Please use the format YYYY-MM-DD." in str(e.value)
+
+    def test_validate_and_save_invalid_favorite_ball_game(self, player):
```

Comment:
Instead of implementing two functions that perform almost identical tests (invalid validation), consider consolidating them into a single function and utilizing `@pytest.mark.parametrize` for better code organization

Reply:
Thanks good tip.